rules-tests/DeadCode/Rector/FunctionLike/NarrowTooWideReturnTypeRector/Fixture/phpdocs.php.inc

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Rector\Tests\DeadCode\Rector\FunctionLike\NarrowTooWideReturnTypeRector\Fixture;
+
+use Rector\Tests\DeadCode\Rector\FunctionLike\NarrowTooWideReturnTypeRector\Source\SomeInterface;
+
+final class PhpDocs
+{
+    /** @return string|int|float */
+    public function foo(): string|int|float
+    {
+        if (rand(0, 1)) {
+            return 'text';
+        }
+
+        return 1000;
+    }
+
+    /**
+     * @param class-string<SomeInterface> $class
+     * @return class-string<SomeInterface>|int
+     */
+    public function bar(string $class): string|int
```

Comment:
please add fixture to skip non-typed based then, to skip this:

```php
    /**
     * @param class-string<SomeInterface> $class
     * @return class-string<SomeInterface>|int
     */
    public function bar($class): string|int
    {
        return $class;
    }
```

Above is marked as `MixedType` on its type, see

https://phpstan.org/r/3e952cb5-c876-4b6e-bfef-89aa5d5b942b

Reply:
I see no reason to do that, this signature is enforced by PHPStan and trying to loop over all the parameter types is only going to unnecessarily complicate the code

Reply:
done

Reply:
Expr returned is not always based on param, see my latest comments :)